js/ui/windowManager.js

```diff
@@ -2214,21 +2226,31 @@ const CodingManager = new Lang.Class({
             return;
         if (focusedWindow === session.actorBuilder.meta_window) {
             if (session.actorApp.meta_window === previousFocused) {
-                this._rotateInActors.push(session.actorBuilder);
-                this._rotateOutActors.push(session.actorApp);
+                // make sure we do not rotate when a rotation is running
+                if (this._rotateInActors.length || this._rotateOutActors.length) {
+                    if (session.buttonBuilder)
+                        session.buttonBuilder.show();
+                    return;
+                }
+                this._prepareAnimate(session.actorApp,
+                                     session.actorBuilder,
+                                     Gtk.DirectionType.LEFT);
                 this._animate(session.actorApp,
                               session.actorBuilder,
                               Gtk.DirectionType.LEFT);
+                session.buttonBuilder.show();
                 session.state = SessionState.BUILDER;
             } else {
                 session.actorBuilder.show();
                 session.buttonBuilder.show();
-                session.actorApp.hide();
```

Comment:
Good catch, I thought this was fixed: What we do now is hide the underlying window in the non-animated-switch case to prevent glitches afterwords when resizing the upper window. Test this by switching to another non session app and back and then resize the session window.

Fixed for both cases and added a comment for that special case.